engine/render/renderer/platform/vulkan/Font.cpp

```diff
@@ -0,0 +1,175 @@
+//
+//  Copyright (c) 2022 Jonathan Moallem (@J-Mo63) & Aryeh Zinn (@Raelr)
+//
+//  This code is released under an unmodified zlib license.
+//  For conditions of distribution and use, please see:
+//      https://opensource.org/licenses/Zlib
+//
+
+#include "Font.h"
+
+#include "Context.h"
+#include "utils/Descriptor.h"
+#include "utils/TypeAdaptor.h"
+#include FT_FREETYPE_H
+
+#include <utils/Defer.h>
+#include <utils/Logging.h>
+
+namespace Siege::Vulkan
+{
+
+FT_Library Font::freetype = nullptr;
+
+Font::Font(const char* filePath)
+{
+    auto device = Context::GetCurrentDevice();
+
+    CC_ASSERT(!FT_New_Face(freetype, filePath, 0, &fontFace), "Failed to load font!")
+
+    uint8_t* buffer = new uint8_t[width * height];
+    defer([buffer]() { delete[] buffer; });
+
+    memset(buffer, 0, sizeof(uint8_t) * width * height);
+
+    texture = Texture2D(filePath,
+                        buffer,
+                        sizeof(uint8_t) * width * height,
+                        width,
+                        height,
+                        Texture2D::Usage::TEX_USAGE_FONT);
+
+    VkSamplerCreateInfo samplerInfo =
+        Utils::Descriptor::SamplerCreateInfo(VK_FILTER_LINEAR,
+                                             VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_EDGE);
+
+    vkCreateSampler(device->GetDevice(), &samplerInfo, nullptr, &info.sampler);
+
+    id = INTERN_STR(filePath);
+
+    for (size_t i = 32; i < 126; i++)
+    {
+        AddChar(static_cast<unsigned char>(i));
+    }
+
+    info.textureInfo = texture.GetInfo();
+}
+
+Font::~Font()
+{
+    Free();
+}
+
+void Font::Free()
+{
+    auto device = Context::GetVkLogicalDevice();
+
+    if (device == nullptr) return;
+
+    vkDestroySampler(device, info.sampler, nullptr);
+
+    texture.Free();
+
+    info = {{}, nullptr};
+}
```

Comment:
Is this safe or useful to make public?

Reply:
We need this as long as static lifetimes continue being a thing. Once we remove the statics in the renderer we can remove this function.